src/Makefile

```diff
@@ -224,7 +224,7 @@ builtins = lang/lang_builtins.ml \
 
 liquidsoap_sources= \
 	tools/log.ml $(tools) converters/audio_converter.ml $(stream) \
-	decoder/decoder.ml request.ml tools/server.ml \
+	URI.ml decoder/decoder.ml request.ml tools/server.ml \
```

Comment:
Have you considered using https://github.com/mirage/ocaml-uri ? Uri parsing can be tricky, using an existing, popular module makes sure some of the quirks have already been considered and fixes..

Reply:
Nice, i'll take a look at that

Reply:
Done with 6085d8c